src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -29364,6 +29625,72 @@ static void GetValue2<T>(T x, out T y)
                 );
         }
 
+        [Fact, WorkItem(39922, "https://github.com/dotnet/roslyn/issues/39922")]
+        public void EnforcedInMethodBody_DoesNotReturn()
```

Comment:
📝 Need tests for `DoesNotReturn` calling another `DoesNotReturn`, e.g.:

```
[DoesNotReturn]
static void Fail() => Fail("message"); // should not warn

[DoesNotReturn]
static void Fail(string message) => throw null!;
```

Reply:
➡️ added test (composes as expected)